managed/services/backup/backup_service_test.go

```diff
@@ -92,84 +95,112 @@ func TestBackup(t *testing.T) {
+				artifactID, err := backupService.PerformBackup(ctx, PerformBackupParams{
+					ServiceID:  pointer.GetString(agent.ServiceID),
+					LocationID: locationRes.ID,
+					Name:       "test_backup",
+				})
+				assert.True(t, errors.Is(err, tc.expectedError))
+				assert.Empty(t, artifactID)
+			})
+		}
+
+		t.Run("success", func(t *testing.T) {
+			versions1 := []agents.Version{
 				{Version: "8.0.25"},
-				{Version: "8.0.24"},
 				{Version: "8.0.25"},
-				{Version: "1.1"},
-			},
-			expectedError: ErrInvalidXtrabackup,
-		},
-		{
-			versions: []agents.Version{
 				{Version: "8.0.25"},
-				{Version: "8.0.24"},
-				{Version: "8.0.24"},
 				{Version: "1.1"},
-			},
-			expectedError: ErrIncompatibleXtrabackup,
-		},
-	} {
-		t.Run(tc.expectedError.Error(), func(t *testing.T) {
-			mockedVersioner.On("GetVersions", *agent.PMMAgentID, softwares).Return(tc.versions, nil).Once()
+			}
+
+			mockedVersioner.On("GetVersions", *agent.PMMAgentID, softwares).Return(versions1, nil).Once()
 			artifactID, err := backupService.PerformBackup(ctx, PerformBackupParams{
 				ServiceID:  pointer.GetString(agent.ServiceID),
 				LocationID: locationRes.ID,
 				Name:       "test_backup",
+				DataModel:  models.PhysicalDataModel,
+				Mode:       models.Snapshot,
 			})
-			assert.True(t, errors.Is(err, tc.expectedError))
-			assert.Empty(t, artifactID)
+			require.NoError(t, err)
+
+			var artifact models.Artifact
+			err = db.SelectOneTo(&artifact, "WHERE id = $1", artifactID)
+			require.NoError(t, err)
+			assert.Equal(t, locationRes.ID, artifact.LocationID)
+			assert.Equal(t, *agent.ServiceID, artifact.ServiceID)
+			assert.EqualValues(t, models.MySQLServiceType, artifact.Vendor)
 		})
-	}
 
-	t.Run("success", func(t *testing.T) {
-		versions1 := []agents.Version{
-			{Version: "8.0.25"},
-			{Version: "8.0.25"},
-			{Version: "8.0.25"},
-			{Version: "1.1"},
-		}
+		mock.AssertExpectationsForObjects(t, mockedJobsService, mockedVersioner, mockedAgentsRegistry)
+	})
 
-		mockedVersioner.On("GetVersions", *agent.PMMAgentID, softwares).Return(versions1, nil).Once()
-		artifactID, err := backupService.PerformBackup(ctx, PerformBackupParams{
-			ServiceID:  pointer.GetString(agent.ServiceID),
-			LocationID: locationRes.ID,
-			Name:       "test_backup",
-			DataModel:  models.PhysicalDataModel,
-			Mode:       models.Snapshot,
-		})
-		require.NoError(t, err)
+	t.Run("mongodb", func(t *testing.T) {
+		agent := setup(t, db.Querier, models.MongoDBServiceType, "test-mongo-service")
 
-		var artifact models.Artifact
-		err = db.SelectOneTo(&artifact, "WHERE id = $1", artifactID)
-		require.NoError(t, err)
-		assert.Equal(t, locationRes.ID, artifact.LocationID)
-		assert.Equal(t, *agent.ServiceID, artifact.ServiceID)
-		assert.EqualValues(t, models.MySQLServiceType, artifact.Vendor)
-	})
+		t.Run("PITR is incompatible with physical backups", func(t *testing.T) {
```

Comment:
And what about other cases?

Reply:
@artemgavrilov, @idoqo what about other cases? Why have you resolved this question?

Reply:
Probably I resolved it accidentally

Reply:
Yeahh covered some more cases, some also exists as API tests (e.g success cases).